src/main/java/com/wolox/training/models/Book.java

```diff
@@ -96,70 +97,79 @@ public String getGenre() {
     }
 
     public void setGenre(String genre) {
-        this.genre = genre;
+        this.genre = Preconditions.checkNotNull(genre,
+                String.format(GuavaPreConditionsMessage.CHECK_NOT_NULL.getMessage(), "genre"));
```

Comment:
Your changes are ok for me, but it would be good to check non-emptyness of the arguments.

 Doing something like: "this.genre = Preconditions.checkArguments(!String.isNullOrEmpty(genre), message);"
 
 Adapt others as needed :call_me_hand: